test/api_test.py

```diff
@@ -79,3 +79,61 @@ def test_unban(self):
         req = responses.calls[0].request
         assert req.url == unban_url
         assert req.method == 'POST'
+
+
+class TestDeviceApi:
+    cli = client.MatrixClient("http://example.com")
+    device_id = "QBUAZIFURK"
+    display_name = "test_name"
+    auth_body = {
+        "auth": {
+            "type": "example.type.foo",
+            "session": "xxxxx",
+            "example_credential": "verypoorsharedsecret"
+        }
+    }
+
+    @responses.activate
+    def test_get_devices(self):
+        get_devices_url = "http://example.com/_matrix/client/r0/devices"
+        responses.add(responses.GET, get_devices_url, body='{}')
+        self.cli.api.get_devices()
+        req = responses.calls[0].request
+        assert req.url == get_devices_url
+        assert req.method == 'GET'
+
+    @responses.activate
+    def test_get_device(self):
+        get_device_url = "http://example.com/_matrix/client/r0/devices/QBUAZIFURK"
+        responses.add(responses.GET, get_device_url, body='{}')
+        self.cli.api.get_device(self.device_id)
+        req = responses.calls[0].request
+        assert req.url == get_device_url
+        assert req.method == 'GET'
+
+    @responses.activate
+    def test_update_device_info(self):
+        update_url = "http://example.com/_matrix/client/r0/devices/QBUAZIFURK"
+        responses.add(responses.PUT, update_url, body='{}')
+        self.cli.api.update_device_info(self.device_id, self.display_name)
+        req = responses.calls[0].request
+        assert req.url == update_url
+        assert req.method == 'PUT'
+
+    @responses.activate
+    def test_delete_device(self):
+        delete_device_url = "http://example.com/_matrix/client/r0/devices/QBUAZIFURK"
+        responses.add(responses.DELETE, delete_device_url, body='{}')
```

Comment:
Can we test that things behave as expected when receiving 401 responses since that's part of the user-interactive auth api?